File: shared/src/main/java/xyz/klinker/messenger/shared/shared_interfaces/IConversationListAdapter.java
package xyz.klinker.messenger.shared.shared_interfaces;

import java.util.List;

import xyz.klinker.messenger.shared.data.SectionType;
import xyz.klinker.messenger.shared.data.model.Conversation;
import xyz.klinker.messenger.shared.data.pojo.ReorderType;

public interface IConversationListAdapter {

    int findPositionForConversationId(long id);
    int getCountForSection(int sectionType);
    boolean removeItem(int position, ReorderType type);

    void notifyItemChanged(int position);
    void notifyItemRangeInserted(int start, int end);
    void notifyItemInserted(int item);

    List<Conversation> getConversations();
    List<SectionType> getSectionCounts();
}

File: shared/src/main/java/xyz/klinker/messenger/shared/shared_interfaces/IConversationListAdapter.kt
package xyz.klinker.messenger.shared.shared_interfaces

import xyz.klinker.messenger.shared.data.SectionType
import xyz.klinker.messenger.shared.data.model.Conversation
import xyz.klinker.messenger.shared.data.pojo.ReorderType

interface IConversationListAdapter {

    val conversations: List<Conversation>
    val sectionCounts: List<SectionType>

    fun findPositionForConversationId(id: Long): Int
    fun getCountForSection(sectionType: Int): Int
    fun removeItem(position: Int, type: ReorderType): Boolean

    fun notifyItemChanged(position: Int)
    fun notifyItemRangeInserted(start: Int, end: Int)
    fun notifyItemInserted(item: Int)
}
